clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2410,16 +2419,17 @@ LogicalResult GetMemberOp::verify() {
 
   // FIXME: currently we bypass typechecking of incomplete types due to errors
   // in the codegen process. This should be removed once the codegen is fixed.
-  if (!recordTy.getBody())
+  if (isIncompleteType(recordTy))
     return mlir::success();
 
   if (recordTy.getMembers().size() <= getIndex())
     return emitError() << "member index out of bounds";
 
   // FIXME(cir): member type check is disabled for classes as the codegen for
   // these still need to be patched.
-  if (!recordTy.isClass() &&
-      recordTy.getMembers()[getIndex()] != getResultTy().getPointee())
+  if (!recordTy.isClass() 
```

Comment:
My only concern here is that we forget the proper enforcement when we fix codegen. Can you factor this a bit out? Example:
```
bool shouldSkipMemberTypeMismatch = recordTy.isClass() || isIncompleteType(recordTy.getMembers()[getIndex()]);
if (!shouldSkipMemberTypeMismatch && recordTy.getMembers()[getIndex()] ...
```